### PR TITLE
feat: implement zunionstore

### PR DESCRIPTION
**Pull request by malandis**
Implements zunionstore and integration tests. Adds a helper class to
validate and return the weights and aggregation options. Adds unit
tests for that helper.

Work towards https://github.com/momentohq/dev-eco-issue-tracker/issues/1033

### CODE REVIEW

File: tests/MomentoRedisClientTest.php [hunk 1]
```diff
@@ -897,6 +897,168 @@ public function testZScoreOnExistingMember(): void
...
+            'a' => 1.0,  // min(1.0 from key1, 2.0 from key2)
+        ];
+        $this->assertEquals($expected, $storedValues, "The stored result in $dst should match the expected min values.");
+    }
+
+    public function testZunionstore_WithInvalidWeights_ShouldThrowException(): void
+    {
+        if (self::$isRedisBackendTest) {
+            $this->markTestSkipped("This test is only for Momento client");
+        }
```
Comment (malandis) [inline tests/MomentoRedisClientTest.php]: I timedboxed detecting what phpredis does in this scenario and could not get it to catch the exception.

File: tests/MomentoRedisClientTest.php [hunk 2]
```diff
@@ -897,6 +897,168 @@ public function testZScoreOnExistingMember(): void
...
+
+        // Attempt zunionstore with invalid weights (mismatched count)
+        self::$client->zunionstore($dst, [$key1, $key2], [1.0]);
+    }
+
+    public function testZunionstore_WithInvalidAggregate_ShouldThrowException(): void
+    {
+        if (self::$isRedisBackendTest) {
+            $this->markTestSkipped("This test is only for Momento client");
+        }
```
Comment (malandis) [inline tests/MomentoRedisClientTest.php]: I timedboxed detecting what phpredis does in this scenario and could not get it to catch the exception.

File: tests/MomentoRedisClientTest.php (hunk 1 above)
Comment (rishtigupta) [reply tests/MomentoRedisClientTest.php]: What does it return in place of an exception?

Comment (malandis) [reply tests/MomentoRedisClientTest.php]: A warning. I've fixed this in [34e14eb](https://github.com/momentohq/momento-php-redis-client/pull/20/commits/34e14eb901eb8f1009003e6869873ea65192f6ce)

**Review by rishtigupta — APPROVED** [review]
:shipit:

**Pull request merged**